File: code/ex836.java
class Solution {
    public boolean isRectangleOverlap(int[] rec1, int[] rec2) {
        int xlo1 = rec1[0], xhi1 = rec1[2], ylo1 = rec1[1], yhi1 = rec1[3];
        int xlo2 = rec2[0], xhi2 = rec2[2], ylo2 = rec2[1], yhi2 = rec2[3];
        
        boolean x = false, y = false;
        if (xlo2 < xhi1 && xlo2 >= xlo1 || xlo1 < xhi2 && xlo1 >= xlo2)
            x = true;
        
        if (ylo2 < yhi1 && ylo2 >= ylo1 || ylo1 < yhi2 && ylo1 >= ylo2)
            y = true;
            
        return x && y;
    }
}

File: code/ex836.cpp
class Solution {
public:
    bool isRectangleOverlap(vector<int>& rec1, vector<int>& rec2) {
        int blx1 = rec1[0], bly1 = rec1[1], trx1 = rec1[2], try1 = rec1[3];
        int blx2 = rec2[0], bly2 = rec2[1], trx2 = rec2[2], try2 = rec2[3];
        return !(blx2 >= trx1 || blx1 >= trx2 || bly1 >= try2 || bly2 >= try1);
    }
};